### PR TITLE
qdel instead of del

### PR DESCRIPTION
**Pull request by gbasood**
All uses of del() replaced with qdel(), with the exception of:
- Things I'd rather not risk breaking (SQL stuff, Regex)
- Admin code, adminbus code

Hopefully I didn't accidentally edit those, might have staged them by accident

### CODE REVIEW

File: code/ATMOSPHERICS/datum_pipeline.dm [hunk 1]
```diff
@@ -16,7 +16,7 @@
 		returnToPool(network)
 	if(air && air.volume) //For the pipeline rebuild next tick
 		temporarily_store_air()
-		del(air)
+		qdel(air)
```
Comment (clusterfack) [inline code/ATMOSPHERICS/datum_pipeline.dm]: air = null after this

File: code/ZAS/ZAS_Turfs.dm [hunk 2]
```diff
@@ -137,7 +137,7 @@
 
 /turf/simulated/Destroy()
 	if(active_hotspot)
-		del(active_hotspot)
+		qdel(active_hotspot)
```
Comment (clusterfack) [inline code/ZAS/ZAS_Turfs.dm]: active_hotspot = null

File: code/__HELPERS/icons.dm [hunk 3]
```diff
@@ -66,7 +66,7 @@ proc
 	text_image.color = AverageColor(atom_icon)
 	text_image.pixel_x = 6
 	text_image.pixel_y = 5
-	del(atom_icon)
+	qdel(atom_icon)
```
Comment (clusterfack) [inline code/__HELPERS/icons.dm]: Okay this cant be qdeleted

**Comment by clusterfack** [comment]
Theres more to check I'll look later

**Comment by clusterfack** [comment]
That solves most of the issues. The primary ones that remain is where it takes to qdel lists, icons, and images in a few places here

**Comment by gbasood** [comment]
Gotta find all the del's that don't use ()'s now woo

File: code/game/gamemodes/endgame/xmas/snow.dm [hunk 4]
```diff
@@ -339,7 +339,7 @@ var/global/list/datum/stack_recipe/snow_recipes = list (
 			new /obj/item/stack/sheet/snow(get_turf(src), 1)
 			new /obj/item/stack/sheet/snow(get_turf(src), 1)
 			new /obj/item/stack/sheet/snow(get_turf(src), 1)
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/game/gamemodes/endgame/xmas/snow.dm]: return after

File: code/game/gamemodes/events/clang.dm [hunk 5]
```diff
@@ -36,7 +36,7 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 			if(clong.density || prob(10))
 				clong.ex_act(2)
 		else
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/game/gamemodes/events/clang.dm]: return after

File: code/game/machinery/OpTable.dm [hunk 6]
```diff
@@ -44,14 +44,14 @@
...
-		del(src)
+		qdel(src)
 
 /obj/machinery/optable/attack_paw(mob/user as mob)
 	if ((M_HULK in usr.mutations))
 		to_chat(usr, text("<span class='notice'>You destroy the operating table.</span>"))
 		visible_message("<span class='warning'>[usr] destroys the operating table!</span>")
 		src.density = 0
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/game/machinery/OpTable.dm]: return

File: code/game/objects/items/candle.dm [hunk 7]
```diff
@@ -41,7 +41,7 @@
 		new/obj/item/trash/candle(src.loc)
 		if(istype(src.loc, /mob))
 			src.dropped()
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/candle.dm]: return

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -55,7 +55,7 @@
 				if(O.reagents.has_reagent("pacid", 1))
 					to_chat(user, "The acid chews through the balloon!")
 					O.reagents.reaction(user)
-					del(src)
+					qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/toys.dm]: return

File: code/game/objects/items/weapons/storage/storage.dm [hunk 9]
```diff
@@ -167,7 +167,7 @@
 
 	New(obj/item/sample as obj)
 		if(!istype(sample))
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/weapons/storage/storage.dm]: return

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 10]
```diff
@@ -74,12 +75,12 @@
 	if(prob(50))
 		var/turf/location = src.loc
 		if (!( istype(location, /turf) ))
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/weapons/tanks/tanks.dm]: return

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 11]
```diff
@@ -253,7 +254,7 @@
 			var/obj/item/device/transfer_valve/TV = src.loc
 			TV.child_ruptured(src, range)
 
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/weapons/tanks/tanks.dm]: return

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 12]
```diff
@@ -263,7 +264,7 @@
 				return
 			T.assume_air(air_contents)
 			playsound(get_turf(src), 'sound/effects/spray.ogg', 10, 1, -3)
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/game/objects/items/weapons/tanks/tanks.dm]: return

File: code/game/objects/structures/signs.dm [hunk 13]
```diff
@@ -33,7 +33,7 @@
 		//var/icon/I = icon('icons/obj/decals.dmi', icon_state)
 		//S.icon = I.Scale(24, 24)
 		S.sign_state = icon_state
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/game/objects/structures/signs.dm]: return

File: code/game/objects/structures/signs.dm [hunk 14]
```diff
@@ -62,7 +62,7 @@
 		S.desc = desc
 		S.icon_state = sign_state
 		to_chat(user, "You fasten \the [S] with your [tool].")
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/game/objects/structures/signs.dm]: return

File: code/modules/games/cards_html.dm [hunk 15]
```diff
@@ -139,7 +142,7 @@
 
 		H.update_icon()
 
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/games/cards_html.dm]: return

File: code/modules/liquid/splash_simulation.dm [hunk 16]
```diff
@@ -65,11 +66,12 @@ obj/effect/liquid
 obj/effect/liquid/New()
 	..()
 	if( !isturf(loc) )
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/liquid/splash_simulation.dm]: return

File: code/modules/mining/mine_items.dm [hunk 17]
```diff
@@ -377,7 +377,7 @@ proc/move_mining_shuttle()
 	if(!(istype(M, /atom/movable)))
 		return
 	if(!(target))
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/mining/mine_items.dm]: return

File: code/modules/liquid/splash_simulation.dm [hunk 18]
```diff
@@ -127,7 +129,7 @@ obj/effect/liquid/proc/apply_calculated_effect()
 	volume += new_volume
 
 	if(volume < LIQUID_TRANSFER_THRESHOLD)
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/liquid/splash_simulation.dm]: return

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 19]
```diff
@@ -168,7 +168,7 @@ emp_act
 				LAssailant = null
 			else
 				LAssailant = user
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/human_defense.dm]: return

File: code/modules/mob/living/carbon/slime/death.dm [hunk 20]
```diff
@@ -14,7 +14,7 @@
 				M1.rabid()
 			var/mob/living/carbon/slime/M2 = new primarytype(loc)
 			M2.rabid()
-			if(src)	del(src)
+			if(src)	qdel(src)
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/slime/death.dm]: return

File: code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm [hunk 21]
```diff
@@ -121,7 +121,7 @@
 		if(amount_grown >= 100)
 			var/spawn_type = pick(spider_types)
 			new spawn_type(src.loc)
-			del(src)
+			qdel(src)
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/hostile/giant_spider/spiderling.dm]: return

File: code/modules/mob/living/simple_animal/hostile/mushroom.dm [hunk 22]
```diff
@@ -79,7 +79,7 @@
 				level_gain = 1
 			M.LevelUp(level_gain)
 		M.health = M.maxHealth
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/hostile/mushroom.dm]: return

File: code/modules/power/engine.dm [hunk 23]
```diff
@@ -8,6 +8,6 @@
 /turf/simulated/floor/engine/blob_act()
 	if(prob(25))
 		ChangeTurf(get_base_turf(src.z))
-		del(src)
+		qdel(src)
```
Comment (clusterfack) [inline code/modules/power/engine.dm]: Just remove this line altogether

**Comment by clusterfack** [comment]
That seems to be all of them

**Comment by PJB3005** [comment]
With this merged we'll have MUCH better performance.
:+1:

**Pull request merged**